### PR TITLE
Test index alignment for checkpointing

### PR DESCRIPTION
**Pull request by navjotk**
This PR primarily adds a test for correct index alignment in `test_checkpointing.py`. 

It also: 
- ~adds a check for missing argument values~
- ~adds a test for the above~

### CODE REVIEW

**Comment by FabioLuporini** [comment]
looks good to me

**Comment by navjotk** [comment]
@FabioLuporini Can you take a look at the travis failure here? Seems to be failing on flake errors in code that I didn't even write - seems like a version issue for flake.

**Comment by FabioLuporini** [comment]
maybe it's pulling a newer flake release, which added E722 and E741? can you update your flake locally and try running it again and see what happens? If you can reproduce the issue locally, feel free to add E722 and E741 to ``setup.cfg``

File: tests/test_checkpointing.py [hunk 1]
```diff
@@ -60,3 +61,64 @@ def test_checkpointed_gradient_test(shape, time_order, space_order):
     m0, dm = example.initial_estimate()
     gradient, rec_data = example.gradient(m0)
     example.verify(m0, gradient, rec_data, dm)
+
+
+@skipif_yask
+def test_index_alignment(const):
```
Comment (FabioLuporini) [inline tests/test_checkpointing.py]: docstring

**Comment by mlange05** [comment]
It looks like this test should be split into at least 3 or 4 independent ones. Is there a reason they are all crammed into the same test function?

**Comment by navjotk** [comment]
I think it's quite a simple test which has multiple asserts on the way, to increase coverage. Is there a reason you think this should be split?

**Comment by FabioLuporini** [comment]
> Is there a reason you think this should be split?

It makes debugging much simpler. If it's feasible to split it, that'd be useful I think

**Comment by navjotk** [comment]
As I said, it's a simple test. I don't see how it can be split.

**Comment by FabioLuporini** [comment]
OK, I'm convinced and happy, this is good to go for me. @navjotk feel free to merge

**Pull request merged**